ethereum_history_api/circuits/lib/src/verifiers/receipt.nr

```diff
@@ -34,18 +34,19 @@ pub fn assert_tx_idx_equals(key_as_rlp: [u8; KEY_LEN], tx_idx: Field) {
     }
 }
 
-fn assert_receipt_proof<MAX_RECEIPT_PROOF_LEN, MAX_RECEIPT_RLP_LEN>(
-    proof: TxReceiptProof,
+pub(crate) fn assert_receipt_proof<MAX_RECEIPT_PROOF_LEN, MAX_RECEIPT_RLP_LEN>(
+    proof: TxReceiptProof<MAX_RECEIPT_RLP_LEN>,
     root: [u8; HASH_LENGTH]
 ) {
-    assert(false, "Not implemented");
+    let trie_proof = TrieProof { key: proof.key, value: proof.value, proof: proof.proof, depth: proof.depth };
```

Comment:
This line makes me wonder why we need following TxReceiptProof like this:
```rs
struct TxReceiptProof {
    key: [u8; MAX_RECEIPT_KEY_LEN],
    value: [u8; MAX_TRIE_NODE_LENGTH],
    proof: [u8; MAX_RECEIPT_PROOF_LEN],
    depth: u64
}
```

When TrieProof definition is:
```rs
struct TrieProof<KEY_LEN, PROOF_LEN, MAX_VALUE_LEN>
{
    key: [u8; KEY_LEN],
    value: [u8; MAX_VALUE_LEN]
    proof: [u8; PROOF_LEN],
    depth: u64,
}
```

Shouldn't we have instead:
```rs
type TxReceiptProof = TrieProof< MAX_RECEIPT_KEY_LEN , MAX_TRIE_NODE_LENGTH, MAX_RECEIPT_PROOF_LEN>
```

And so the line above would be unnecessary.

Reply:
Added commit that replaces `TxReceiptProof` with `ReceiptProof` from `noir-trie-proofs` library. The same can be done for transaction